plugin/src/main/kotlin/tanvd/grazi/grammar/GrammarChecker.kt

```diff
@@ -7,30 +7,35 @@ import tanvd.kex.*
 
 class GrammarChecker(private val ignoreChar: LinkedSet<(CharSequence, Char) -> Boolean> = LinkedSet(),
                      private val replaceChar: LinkedSet<(CharSequence, Char) -> Char?> = LinkedSet(),
-                     private val ignoreToken: LinkedSet<(String) -> Boolean> = LinkedSet()) {
+                     private val ignoreToken: LinkedSet<(String) -> Boolean> = LinkedSet(),
+                     private val afterResultReplace: LinkedSet<(String) -> String> = LinkedSet()) {
 
     constructor(checker: GrammarChecker, ignoreChar: LinkedSet<(CharSequence, Char) -> Boolean> = LinkedSet(),
                 replaceChar: LinkedSet<(CharSequence, Char) -> Char?> = LinkedSet(),
-                ignoreToken: LinkedSet<(String) -> Boolean> = LinkedSet())
-            : this(LinkedSet(checker.ignoreChar + ignoreChar), LinkedSet(checker.replaceChar + replaceChar), LinkedSet(checker.ignoreToken + ignoreToken))
+                ignoreToken: LinkedSet<(String) -> Boolean> = LinkedSet(),
+                afterResultReplace: LinkedSet<(String) -> String> = LinkedSet())
```

Comment:
Note that you must replace with equal sized string to maintain consistency. It is a very implicit requirement.

Reply:
Outdated